### PR TITLE
WIP: Fix bad network until a project is opened

### LINKED ISSUE #2610: [BUG]: Desktop app shows bad network until a project is opened
### Describe the bug

Presumably because the engine connection is not initialized until the first project is opened, the network health indicator shows a nebulous problem state until the user enters their first file. We will want tauri tests for this eventually.

Some solutions in my order of preference:
1. Connect to the engine on app start up. I think this would have the knock-on effect of making first project load faster too.
2. Make the network health indicator only check the validity of network connected status until the engine connection is attempted.
3. Hide the network connection on home page

### Steps to Reproduce

1. Open the desktop app

### Expected Behavior

The network health indicator should only show a problem if there is a problem the user needs to address or that is otherwise interfering with app use.

### Screenshots and Recordings

Screenshot 2024-06-05 at 1 19 43 PM

tauri tests playwright tests electron tests test

### Desktop OS

Any

### Browser

None (desktop version only)

### Version

v0.22.0

### Additional Context

@jessfraz this one might be borderline not high-priority: it's high visibility but low user impact.

### PR DESCRIPTION
**Pull request by pierremtb**
Will fix #2610 and add tauri e2e test to check on the network indicator. 

Merge #2659 first

### CODE REVIEW

File: src/Router.tsx [hunk 1]
```diff
@@ -123,9 +123,12 @@ const router = createBrowserRouter([
...
           <Auth>
-            <Outlet />
-            <Home />
-            <CommandBar />
+            <ModelingMachineProvider>
+              <Outlet />
+              <Home />
+              <CommandBar />
+            </ModelingMachineProvider>
+            <WasmErrBanner />
```
Comment (pierremtb) [inline src/Router.tsx:131]: Noticed that this work only when you open the app signed in. Works after a refresh after signing in, which is probably why the test fails

Comment (pierremtb) [reply src/Router.tsx:131]: Yeah [forcing a refresh after auth](https://github.com/KittyCAD/modeling-app/pull/2661/files#diff-b104af957bda276a1f3e77c8c4bf09dcce1e262c0667c0ef10008f8a27e546c3R73) makes the test happy.

@lf94 @franknoirot The way I tried to enable the engine connection on the home page here was a complete shot in the dark. Let me know how you'd want to proceed

Comment (pierremtb) [reply src/Router.tsx:131]: Also, sorry for the dirty diff here. We should merge https://github.com/KittyCAD/modeling-app/pull/2659 first

**Comment by lf94** [comment]
I can check this when Im home in 3hr. Still wip?

**Comment by pierremtb** [comment]
Well yeah still wip cause I have no idea if it’s the right solution, nor does it work first time after sign inOn Jun 16, 2024, at 4:59 PM, 49fl ***@***.***> wrote:﻿
I can check this when Im home in 3hr. Still wip?

**Comment by lf94** [comment]
Ah ok, it should work after first sign in I think to move forward on this but good initial work

**Comment by franknoirot** [comment]
Just merged #2659!

**Comment by pierremtb** [comment]
@franknoirot Thank you! Updating

**Comment by Irev-Dev** [comment]
Is this work still ongoing, or maybe we figured it was the wrong approach?

**Comment by pierremtb** [comment]
@Irev-Dev Not sure about the approach but I think we should keep the test. Kicking a branch update now

**Comment by jessfraz** [comment]
closed this was fixed elesewhere

**Pull request closed without merging**